src/core/core/rpc_engine.cpp

```diff
@@ -144,7 +144,12 @@ namespace dsn {
                 req->server_address.group_address()->leader_forward();
             }
 
-            call->set_delay(delay_ms);
+            // the "delay_ms" is simulator's net_delay_milliseconds,
+            // the "call->delay_milliseconds()" is user specified delay time to dispatch response task,
```

Comment:
Btw, the old logic to use response_task->set_delay() is not correct - it won't delay the rpc call but only the rpc response delivery.